kernel/src/process/clone.rs

```diff
@@ -271,6 +273,12 @@ fn clone_child_task(
         )
     }
 
+    // Check CLONE_PARENT flag.
+    // The CLONE_PARENT flag has no real effect when cloning a new thread.
+    // A newly created thread always has the same parent process as the caller,
+    // no matter whether the flag is set. However, we still need to perform the required check.
```

Comment:
But we've performed some checks here:
https://github.com/asterinas/asterinas/blob/66356e133db8eee843bdffae29303e3e62fc4e7e/kernel/src/process/clone.rs#L272-L284

So if we don't specify `CLONE_PARENT`, can we specify `exit_signal` for `CLONE_THREAD`?

Reply:
> So if we don't specify CLONE_PARENT, can we specify exit_signal for CLONE_THREAD?

Yes, we also cannot specify an exit_signal when using CLONE_THREAD. After further experiments I discovered this is specific to the clone3 syscall: clone3 enforces exit-signal checks, while the clone syscall does not. Accordingly, I moved the validation into the clone3 argument checks.

I also consolidated all argument-validation logic related to CloneArgs into CloneArgs::check, so these checks are performed in one central place.